datafusion/optimizer/src/decorrelate_dependent_join.rs

```diff
@@ -393,19 +404,31 @@ impl DependentJoinDecorrelator {
             }
         }
 
-        for col in node
-            .correlated_columns
-            .iter()
-            .map(|info| info.col.clone())
-            .unique()
-        {
-            let raw_name = col.flat_name().replace('.', "_");
+        // TODO: Revisit this delim join condition
```

Comment:
i tested around this with multiple queries, not sure if any other case still cause problems